scheduler/src/main/scala/com.gu.media.scheduler/SchedulerLambda.scala

```diff
@@ -0,0 +1,79 @@
+package com.gu.media.scheduler
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
+import com.gu.contentatom.thrift.atom.media.PrivacyStatus
+import com.gu.media.CapiAccess
+import com.gu.media.lambda.LambdaBase
+import com.gu.media.logging.Logging
+import com.gu.media.youtube.{YouTubeAccess, YouTubeVideos}
+import play.api.libs.json.{JsArray, JsValue}
+
+import scala.annotation.tailrec
+import scala.util.control.NonFatal
+
+class SchedulerLambda extends RequestHandler[Unit, Unit]
+  with LambdaBase
+  with Logging
+  with CapiAccess
+  with YouTubeAccess
+  with YouTubeVideos {
+
+  def scheduleInParallel = true // disabled in unit tests
+
+  override def handleRequest(input: Unit, context: Context): Unit = {
+    val now = Instant.now()
+    val oneDayAgo = Instant.now().minus(1, ChronoUnit.DAYS)
+    val assets = getVideosFromScheduledAtoms(1, 100, oneDayAgo, now, Set.empty).filter(isManagedVideo)
+
+    val toLaunch = if(scheduleInParallel) { assets.par } else { assets }
```

Comment:
How many videos are we talking about? Just tried a couple requests and got 0 results all the time. This may be a premature optimization.

Reply:
Ok, this is a leftover from my refactor of a previous version of a similar lambda. The lambda used to run for all the existing atoms (before filtering by scheduledLaunch was possible in CAPI), so that's probably why the optimisation was there? Now that we are filtering and getting fewer results this shouldn't be needed anymore.